test/e2e/node/pre_stop.go

```diff
@@ -193,7 +193,8 @@ var _ = SIGDescribe("PreStop", func() {
 
 		ginkgo.By("verifying the pod is running while in the graceful period termination")
 		result := &v1.PodList{}
-		err = wait.Poll(time.Second*5, time.Second*60, func() (bool, error) {
```

Comment:
this poll loop is incorrect, since it doesn't retry, it should be rewrite for retrying if any of the external conditions fail
```go
		ginkgo.By("verifying the pod is running while in the graceful period termination")
		result := &v1.PodList{}
		err = wait.Poll(time.Second*5, time.Second*60, func() (bool, error) {
			client, err := e2ekubelet.ProxyRequest(f.ClientSet, pod.Spec.NodeName, "pods", ports.KubeletPort)
                        if err != nil {
               			framework.Logf( "failed to get the pods of the node")
                                return false, nil
                         }
                	err = client.Into(result)
                        if err != nil {
               			framework.Logf( "failed to parse the pods of the node")
                                return false, nil
                         }

			for _, kubeletPod := range result.Items {
				if pod.Name != kubeletPod.Name {
					continue
				} else if kubeletPod.Status.Phase == v1.PodRunning {
					framework.Logf("pod is running")
					return true, nil
				}
			}
			return false, nil
		})

		framework.ExpectNoError(err, "validate-pod-is-running")
```

Reply:
Will update

Reply:
@aojea I had updated the logic. What we are doing here is checking if pod is running even after deleting the pod before gracefulTermination period ends as the test name suggests, as current execution doesnt check that.

Reply:
@aojea but here even if we don't check whether err != nil, it would still `return false, nil` right?
It would still let us decrease time it takes in the loop, so I think we can also decrease the interval time? Updated in the PR

Reply:
I was just pointing out the logic was wrong because if you return the error it never retries, hence the wait.Poll loop was useless.
Regarding the test logic, it is better if sig-node folks chime in